rust_team_data/src/email_encryption.rs

```diff
@@ -0,0 +1,111 @@
+//! This module implements the encryption scheme used to safely include private email addresses in
+//! the team repository. It generates encrypted content that looks like this:
+//!
+//! encrypted+a371d1d65dcc697d000826e4bd59dcbfb16ad73eceb1139caaaab795737bb3a54403f0ec69f3d40c8965a5f4bc52f67c94ec538a8a2fea710cabdaaf27661f9a62b0121db7d229@rust-lang.invalid
+//!
+//! The hex-encoded part of the email address is a concatenation of a 12-byte random nonce and the
+//! AES256-GCM-encrypted email address. Utilities are provided to both encrypt and decrypt.
+
+use aes_gcm::aead::{generic_array::GenericArray, Aead, NewAead};
+use aes_gcm::Aes256Gcm;
```

Comment:
Was there some particular reason to choose this particular crate and cipher?

Reply:
AES256-GCM is a secure AEAD cipher, and it's definitely good for our use case. There is no particular reason why I chose it compared to other secure AEAD ciphers.

The crate is part of the [RustCrypto project](https://github.com/RustCrypto/) and received [a security audit](https://research.nccgroup.com/2020/02/26/public-report-rustcrypto-aes-gcm-and-chacha20poly1305-implementation-review/). In general I find the maintenance policy of RustCrypto crates to be better for downstream users compared to *ring*.

Reply:
We discussed a little on Zulip, and after some consideration it felt like we should probably switch to https://crates.io/crates/chacha20poly1305 - some random googling seems to say that AES256-GCM is not the best thing to use (though likely it doesn't matter for us); libsodium's docs also recommend the use of this primitive.

https://doc.libsodium.org/secret-key_cryptography/aead/chacha20-poly1305/xchacha20-poly1305_construction

Reply:
Switched algorithm.